ext/stackprof/stackprof.c

```diff
@@ -639,6 +682,22 @@ stackprof_atfork_child(void)
     stackprof_stop(rb_mStackProf);
 }
 
+long int timeval_to_usec(struct timeval *diff) {
+    return MICROSECONDS_IN_SECOND * diff->tv_sec + diff->tv_usec;
+}
+
+long int diff_timevals_usec(struct timeval *start, struct timeval *end) {
```

Comment:
(I adapted this from an algorithm at https://www.guyrutenberg.com/2007/09/22/profiling-code-using-clock_gettime/, and tested it with this repl: https://repl.it/@BenBuckman/SanityCheck-Timevals)